nntrainer/src/neuralnet.cpp

```diff
@@ -280,11 +280,11 @@ sharedConstTensor NeuralNetwork::forwarding(sharedConstTensor input) {
 /**
  * @brief     forward propagation using layers object which has layer
  */
-sharedConstTensor NeuralNetwork::forwarding(sharedConstTensor input,
-                                            sharedConstTensor label) {
-  sharedConstTensor X;
+sharedConstTensors NeuralNetwork::forwarding(sharedConstTensors input,
+                                             sharedConstTensors label) {
+  sharedConstTensors X;
 
-  if (input->getDim().batch() > batch_size)
+  if (input[0]->getDim().batch() > batch_size)
```

Comment:
as the input is coming from outside, lets also check that the size of the vector matches with number of inputs for the first layer. It is sufficient to be done only for the first layer, as the input from user can be erroneous but intermediate layers can be trusted and have been verified.
Same for the label going to the backwarding.

Reply:
I don't think multiple input for layers allowed even if it is input layer. IMO, if someone want to use multiple tensor, then he have to use concatenate or add layer first after set individual input layer.

I think We need to kind of tensors class for better handling but it can be separate pr though. Let's assume that this pr is to enable skip connection.

Reply:
Then in the later PR, we can have a check on the count of the number of input tensors and the number of input layers.